marklogic-data-hub/src/test/java/com/marklogic/hub/hubcentral/migration/AllStepsReferencedTest.java

```diff
@@ -0,0 +1,429 @@
+package com.marklogic.hub.hubcentral.migration;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+//import com.marklogic.client.document.JSONDocumentManager;
+import com.marklogic.client.eval.EvalResult;
+import com.marklogic.client.eval.EvalResultIterator;
+import com.marklogic.hub.*;
+import com.marklogic.hub.dataservices.FlowService;
+import com.marklogic.hub.flow.Flow;
+import com.marklogic.hub.flow.FlowInputs;
+import com.marklogic.hub.flow.FlowRunner;
+import com.marklogic.hub.flow.RunFlowResponse;
+import com.marklogic.hub.impl.FlowManagerImpl;
+import com.marklogic.hub.impl.MappingManagerImpl;
+import com.marklogic.hub.mapping.Mapping;
+import com.marklogic.hub.step.RunStepResponse;
+import com.marklogic.hub.step.StepDefinition;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ClassPathResource;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class AllStepsReferencedTest extends AbstractHubCoreTest {
```

Comment:
There seems to be a lot of overlap (and duplication) between this test and FlowMigratorTest. I think it would be better to have one test  class in which a project can be migrated, the artifacts verified and then run.

Reply:
I don't mind a separate test class, as long as we have separate scenarios (in fact, complex scenarios often warrant their own test class with a single test method). I agree that duplication of assertions on ingestions and mapping steps can be removed. But I don't mind a separate class.